### PR TITLE
generate boundary with token_hex

### PR DESCRIPTION
**Pull request by trim21**
# Summary

# Checklist

- [ ] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

File: starlette/responses.py [hunk 1]
```diff
@@ -404,7 +404,7 @@ async def _handle_multiple_ranges(
         file_size: int,
         send_header_only: bool,
     ) -> None:
-        boundary = "".join(random_choices("abcdefghijklmnopqrstuvwxyz0123456789", k=13))
+        boundary = token_hex(20)
```
Comment (Kludex) [inline starlette/responses.py]: Why was it increased to 20?

Comment (trim21) [reply starlette/responses.py]: just A random number I picked. len(boundary) is actually 40 now

Comment (abersheeran) [reply starlette/responses.py]: According to the specification, this value can be between 1 and 70. But is 40 too long?

**Review by abersheeran — APPROVED** [review]
Elegant modification.

File: starlette/responses.py (hunk 1 above)
Comment (graingert) [reply starlette/responses.py]: \>96 bits of entropy is recommended so only 13 bytes is needed

Comment (abersheeran) [reply starlette/responses.py]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type#directives

Comment (Kludex) [reply starlette/responses.py]: Can we drop a bit the number then?

Comment (Kludex) [reply starlette/responses.py]: @trim21 Can we drop the number a bit so I can include this on the next release?

Comment (trim21) [reply starlette/responses.py]: to what? 13?

Comment (Kludex) [reply starlette/responses.py]: Yep. Maybe we can add a comment about the choice, to avoid future confusion/questions?

Comment (trim21) [reply starlette/responses.py]: > >96 bits of entropy is recommended so only 13 bytes is needed

where does this recommendation comes from?

Comment (graingert) [reply starlette/responses.py]: https://github.com/whatwg/html/issues/6424#issuecomment-826036018
https://andreubotella.github.io/multipart-form-data/#serializing
https://github.com/andreubotella/multipart-form-data/blob/main/index.bs#L32

Also I misremembered the recommendation it's 95 bits, so 12 bytes would be sufficient

**Review by graingert — APPROVED** [review]

File: starlette/responses.py [hunk 2]
```diff
@@ -404,7 +404,9 @@ async def _handle_multiple_ranges(
         file_size: int,
         send_header_only: bool,
     ) -> None:
-        boundary = "".join(random_choices("abcdefghijklmnopqrstuvwxyz0123456789", k=13))
+        # in firefox and chrome, they use boundary with 95-96 bits entropy,
+        # that's roughly 13 bytes
```
Comment (Kludex) [inline starlette/responses.py]:
```suggestion
        # In firefox and chrome, they use boundary with 95-96 bits entropy (that's roughly 13 bytes).
```

**Pull request merged**